### PR TITLE
[SPARK-44334][SQL][UI] Status in the REST API response for a failed DDL/DML with no jobs should be FAILED rather than COMPLETED

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR fixes the bug, which responds COMPLETED for execution w/ error and w/o jobs in SqlResource, and makes it consistent with AllExecutionsPage

### Why are the changes needed?

bugfix

### Does this PR introduce _any_ user-facing change?

yes, but bugfix

### How was this patch tested?

new unit test

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @ulysses-you thanks

**Comment by yaooqinn** [comment]
also cc @dongjoon-hyun @HyukjinKwon @sarutak

**Comment by HyukjinKwon** [comment]
cc @gengliangwang too

**Comment by sarutak** [comment]
@yaooqinn 
Could you elaborate the issue in the description?
I understand the issue is that, when a DDL or DML which has no jobs is executed and then fail, the status of the corresponding execution shown in `/api/v1/applications/<app-id>/sql` is marked as `COMPLETED` even though it's marked as `FAILED` in `AllExecutionsPage`.
If so, I think it's better to add a UT which ensures that the response of the REST API is as expected with such a DDL/DML.

File: sql/core/src/test/scala/org/apache/spark/status/api/v1/sql/SqlResourceSuite.scala [hunk 1]
```diff
@@ -204,4 +203,25 @@ class SqlResourceSuite extends SparkFunSuite with PrivateMethodTester {
     assert(wholeStageCodegenId == Some(1))
   }
 
+  test("Status of execution w/ error and w/o jobs shall be FAILED not COMPLETED") {
```
Comment (sarutak) [inline sql/core/src/test/scala/org/apache/spark/status/api/v1/sql/SqlResourceSuite.scala:206]: Could you prepend `SPARK-44334: ` to the test name?

**Comment by sarutak** [comment]
I'm wondering if we can give a better name for this PR.
I'm not a native English speaker but how about `Status in the REST API response for a failed DDL/DML with no jobs should be FAILED rather than COMPLETED`

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/status/api/v1/sql/SqlResourceSuite.scala (hunk 1 above)
Comment (yaooqinn) [reply sql/core/src/test/scala/org/apache/spark/status/api/v1/sql/SqlResourceSuite.scala]: addressed, and I add a new test case with a DDL.

**Review by sarutak — APPROVED** [review]

**Comment by yaooqinn** [comment]
All tests passed except a py-linter job https://github.com/yaooqinn/spark/actions/runs/5526869682/jobs/10082068689 not related.

merged to master. thanks @cloud-fan @sarutak and @HyukjinKwon

**Pull request closed without merging**